### PR TITLE
Replace tslint with eslint and prettier

### PR DESCRIPTION
**Pull request by BrendanAnnable**

### CODE REVIEW

**Comment by TrentHouliston** [comment]
Can we set the line width to 120 characters rather than 80?

**Comment by JosephusPaye** [comment]
> Can we set the line width to 120 characters rather than 80?

120 might be a bit too much, and [prettier works best](https://prettier.io/docs/en/options.html#print-width) with a shorter line length since it'll try to fit as much as possible in the given width.

Maybe 100? Don't know how the result of that will look.

File: src/client/components/vision/camera/view_model.ts [hunk 1]
```diff
@@ -430,10 +466,22 @@ export class CameraViewModel {
...
+    mat4.z.y,
+    mat4.t.y,
+    mat4.x.z,
+    mat4.y.z,
+    mat4.z.z,
+    mat4.t.z,
+    mat4.x.t,
+    mat4.y.t,
+    mat4.z.t,
+    mat4.t.t,
```
Comment (BrendanAnnable) [inline src/client/components/vision/camera/view_model.ts]: 😢

File: src/client/components/vision2/image.ts [hunk 2]
```diff
@@ -38,8 +38,8 @@ export enum ImageFormat {
   Y8__, // Monochrome
 }
 
-export type BayerImageFormat
-  = ImageFormat.BGGR
+export type BayerImageFormat =
+  | ImageFormat.BGGR
```
Comment (BrendanAnnable) [inline src/client/components/vision2/image.ts]: 🤨

File: src/client/components/visual_mesh/camera/view_model.ts [hunk 3]
```diff
@@ -107,65 +110,70 @@ export class CameraViewModel {
...
-    for (let i = 0; i < nElem; i++) {
-      const ni = i * 6
-      if (neighbours[ni + 0] < nElem) {
-        if (neighbours[ni + 2] < nElem) {
-          triangles.push(i, neighbours[ni + 0], neighbours[ni + 2])
-        }
-        if (neighbours[ni + 1] < nElem) {
-          triangles.push(i, neighbours[ni + 1], neighbours[ni + 0])
+  private meshGeometry = createTransformer(
+    (mesh: VisualMesh): BufferGeometry => {
```
Comment (BrendanAnnable) [inline src/client/components/visual_mesh/camera/view_model.ts]: 🤨

File: src/client/components/vision/camera/view_model.ts (hunk 1 above)
Comment (TrentHouliston) [reply src/client/components/vision/camera/view_model.ts]: Are there comment blocks you can do to disable prettier for a few lines?

File: src/client/components/vision2/image.ts (hunk 2 above)
Comment (TrentHouliston) [reply src/client/components/vision2/image.ts]: What has happened here?

File: src/client/components/vision/camera/view_model.ts (hunk 1 above)
Comment (BrendanAnnable) [reply src/client/components/vision/camera/view_model.ts]: 4df099ea

File: src/client/components/vision2/image.ts (hunk 2 above)
Comment (BrendanAnnable) [reply src/client/components/vision2/image.ts]: its valid syntax apparently

**Comment by JosephusPaye** [comment]
🎉

**Pull request merged**